source/ShopPanel.cpp

```diff
@@ -64,11 +64,18 @@ void ShopPanel::Step()
 	if(player.Ships().size() > 1)
 		DoHelp("multiple ships");
 	// Perform autoscroll to bring item details into view.
-	if(scrollDetailsIntoView && selectedBottomY > 0.)
+	if(scrollDetailsIntoView && mainDetailHeight > 0)
 	{
-		double offY = Screen::Bottom() - selectedBottomY;
-		if(offY < 0.)
-			DoScroll(max(-30., offY));
+		int mainTopY = Screen::Height() / -2;
+		int mainBottomY = Screen::Height() / 2 - 40;
```

Comment:
`Screen::Height() / -2` is the same as `Screen::Top()`, but the latter would be easier to read.

Reply:
Done.